src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php

```diff
@@ -842,4 +846,25 @@ private function getOrderProducts(OrderId $orderId, string $productsOrder): Orde
             GetOrderProductsForViewing::all($orderId->getValue(), $productsOrder)
         );
     }
+
+    /**
+     * If there is no valid customer attached to the order, the customer must have been deleted
+     * from the database. We then create a fake customer object, using the invoice address data
+     * and cart language.
+     *
+     * @param Order $order Order object
+     * @param OrderInvoiceAddressForViewing $invoiceAddress Invoice address information
+     * @param Customer $customer Customer object
+     *
+     * @return void
+     */
+    private function buildFakeCustomerObject(Order $order, OrderInvoiceAddressForViewing $invoiceAddress, Customer $customer): void
+    {
+        $cart = new Cart($order->id_cart);
+
+        $customer->firstname = $invoiceAddress->getFirstName();
+        $customer->lastname = $invoiceAddress->getLastName();
+        $customer->email = '';
+        $customer->id_lang = $cart->getAssociatedLanguage()->getId();
```

Comment:
```suggestion
        $customer->id_lang = $cart->getAssociatedLanguage()->getId();
        
        return $customer
```

Reply:
Shouldn't `$customer->is_guest` be set to `true` to avoid issues?

Reply:
Good question, I think you're right, the customer is now a guest as the account doesn't exist anymore.